cmd/recv.go

```diff
@@ -199,6 +199,18 @@ func recvAction(cmd *cobra.Command, args []string) {
 					bail("Read zip error: %s", err)
 				}
 
+				// calculate the uncompressed size of the contents of the zip
+				var actualUncompressedSize uint64
+				var fileCount int
+				for _, f := range zr.File {
+					actualUncompressedSize += f.FileHeader.UncompressedSize64
+					fileCount++
+				}
+				if msg.UncompressedBytes64 != int64(actualUncompressedSize) ||
+					msg.FileCount != fileCount {
```

Comment:
I think i'd prefer to reject transfers where `msg.UncompressedBytes64 < actualUncompressedSize` or `msg.FileCount < fileCount`. This will maintain the safety property but will still allow for transfers if the sender has an off by one bug in how it calculates its offer.